python/lsst/sconsUtils/state.py

```diff
@@ -378,8 +380,8 @@ def ClassifyCc(context):
     # Set the optimization level.
     #
     if env['opt']:
-        env["CCFLAGS"] = [o for o in env["CCFLAGS"] if not re.search(r"^-O(\d|s)$", o)]
-        env.MergeFlags('-O%d' % int(env['opt']))
+        env["CCFLAGS"] = [o for o in env["CCFLAGS"] if not re.search(r"^-O(\d|s|g)$", o)]
```

Comment:
`-Ofast` will mess this up if we ever use it. Not that I'm suggesting we support `fast` at the moment.